### PR TITLE
Update react-native and react-relay version

### PR DESCRIPTION
**Pull request by alvinthen**
Modified packages according to https://github.com/facebook/relay/issues/26#issuecomment-168523524

### CODE REVIEW

**Comment by lenaten** [comment]
awesome man, thanks!

e.g. any suggestion why the collection example doesn't work on android?

**Comment by alvinthen** [comment]
Seems like this is related
https://github.com/facebook/react-native/issues/5262
I've used the master branch of react-native, let me try again patching over
a stable build
